test/samplers/test_kennard_stone.py

```diff
@@ -14,11 +16,109 @@ class Test_kennard_stone(unittest.TestCase):
 
     @classmethod
     def setUpClass(self):
-        return
+        self.X = np.array(
+            [
+                [0, 0, 0],
+                [0, 1, 0],
+                [1, 1, 1],
+            ]
+        )  # KS will sample two furthest for train
+        self.y = np.array([1, 2, 3])
+        self.labels = np.array(
+            [
+                "one",
+                "two",
+                "three",
+            ]
+        )
+
+    def test_kennard_stone_sample(self):
+        """Use kennard stone in tts and verify results """
+        with self.assertWarns(ImperfectSplittingWarning):
+            (
+                X_train,
+                X_test,
+                y_train,
+                y_test,
+                labels_train,
+                labels_test,
+            ) = train_test_split(
+                self.X,
+                self.y,
+                labels=self.labels,
+                test_size=0.25,
+                train_size=0.75,
+                sampler="kennard_stone",
+            )
+        # test that the known arrays equal the result from above
+        self.assertIsNone(
+            np.testing.assert_array_equal(
+                X_train,
+                np.array([[0, 0, 0], [1, 1, 1]]),
+            ),
+            "Train X incorrect.",
+        )
+        self.assertIsNone(
+            np.testing.assert_array_equal(
+                X_test,
+                np.array([[0, 1, 0]]),
+            ),
+            "Test X incorrect.",
+        )
+        self.assertIsNone(
+            np.testing.assert_array_equal(
+                y_train,
+                np.array([1, 3]),
+            ),
+            "Train y incorrect.",
+        )
+        self.assertIsNone(
+            np.testing.assert_array_equal(
+                y_test,
+                np.array([2]),
+            ),
+            "Test y incorrect.",
+        )
+        self.assertIsNone(
+            np.testing.assert_array_equal(
+                labels_train,
+                np.array(["one", "three"]),
+            ),
+            "Train labels incorrect.",
+        )
+        self.assertIsNone(
+            np.testing.assert_array_equal(
+                labels_test,
+                np.array(["two"]),
+            ),
+            "Test labels incorrect.",
+        )
 
     def test_kennard_stone(self):
```

Comment:
Sorry to be overly pedantic. Would it make sense to place this test first in the class so that it's run first? Seems like it's the most basic unit test to check before running our wrapper on it.

Reply:
This test will actually be run first by default since unittest executes tests alphabetically, but I will move it first in the file to make that clearer. Agreed that it makes more sense.